[OSDOCS-4789]: Machine deletion hooks (#55175)
openshift/openshift-docs: modules/machine-delete.adoc

.Procedure

. View the machines that are in the cluster by running the following command:
+
[source,terminal]
----
$ oc get machine -n openshift-machine-api
----
+
The command output contains a list of machines in the `<clusterid>-<role>-<cloud_region>` format.

. Identify the machine that you want to delete.

. Delete the machine by running the following command:
+
[source,terminal]
----
$ oc delete machine <machine> -n openshift-machine-api
----
+
[IMPORTANT]
====

> By default, the machine controller tries to drain the node that is backed by the machine until it succeeds. In some situations, such as with a misconfigured pod disruption budget, the drain operation might not be able to succeed in preventing the machine from being deleted. You can skip draining the node by annotating `machine.openshift.io/exclude-node-draining` in a specific machine.

Reviewer: > the drain operation might not be able to succeed in preventing the machine from being deleted.

I'm not following this part of the sentence, think something got mixed up here

Author: hm yeah - iiuc, maybe something like this?

```suggestion
By default, the machine controller tries to drain the node that is backed by the machine until it succeeds. If the drain operation fails, the machine cannot be deleted. You can skip draining the node by annotating `machine.openshift.io/exclude-node-draining` in a specific machine.
```

I don't know how important it is to give the example of what would cause the failure (bad PDB) - what do you think?

Reviewer: > If the drain operation fails, the machine cannot be deleted.

Maybe cannot proceed removing the machine? 

And yes, and example of PDB blocking the drain would be helpful I think

====
+
If the machine that you delete belongs to a machine set, a new machine is immediately created to satisfy the specified number of replicas.

> //I think this is true for both compute and control plane machine sets as stated, but should be verified.

Reviewer: Yes it is correct
